src/librustdoc/clean/types.rs

```diff
@@ -1966,11 +1967,64 @@ crate struct Static {
 }
 
 #[derive(Clone, PartialEq, Eq, Hash, Debug)]
-crate struct Constant {
-    crate type_: Type,
-    crate expr: String,
-    crate value: Option<String>,
-    crate is_literal: bool,
+crate enum Constant {
+    /// Typed constant value.
+    TyConst { type_: Type, expr: String },
+    /// A constant (expression) that’s not an item or associated item. These are usually found
+    /// nested inside types (e.g., array lengths) or expressions (e.g., repeat counts), and also
+    /// used to define explicit discriminant values for enum variants.
+    Generic { type_: Type, body: BodyId },
```

Comment:
Generic seems like a bad name for this. Maybe "Anonymous" is better?

Reply:
The original type is [AnonConst](https://doc.rust-lang.org/nightly/nightly-rustc/rustc_hir/hir/struct.AnonConst.html) so you're right, it'd be much better!